crates/runtime/src/instance.rs

```diff
@@ -140,6 +143,9 @@ pub struct Instance {
     /// seems not too bad.
     vmctx_self_reference: SendSyncPtr<VMContext>,
 
+    #[cfg(feature = "valgrind")]
+    pub(crate) valgrind_state: Option<Valgrind>,
```

Comment:
Let's add a TODO here to support multiple memories; for now we have just one `valgrind_state` and it corresponds to memory 0.